src/renderer.c

```diff
@@ -725,18 +738,18 @@ void ren_draw_rect(RenSurface *rs, RenRect rect, RenColor color) {
                          rect.height * surface_scale };
 
   if (color.a == 0xff) {
-    uint32_t translated = SDL_MapRGB(surface->format, color.r, color.g, color.b);
-    SDL_FillRect(surface, &dest_rect, translated);
+    uint32_t translated = SDL_MapSurfaceRGB(surface, color.r, color.g, color.b);
+    SDL_FillSurfaceRect(surface, &dest_rect, translated);
   } else {
     // Seems like SDL doesn't handle clipping as we expect when using
     // scaled blitting, so we "clip" manually.
```

Comment:
Might be worth it to check if this got fixed.

Reply:
The code for this did not change in SDL3 (not for the last 5 years).

I looked at the code and I'm not sure what you meant by "doesn't handle clipping as we expect".
It does seemingly clip the rectangles when the coordinates are off-screen,
and your commit was from 2 years ago.

Reply:
I *am* talking about my code. At the time I tried to leave the clipping completely to SDL, but it didn't work as expected. I don't remember what was wrong tho.
This is why I'm hoping SDL3 fixed the issue and we can drop the extra code.

Reply:
Yes, but looking at the code it seems like the kind of clipping we do explicitly is done by SDL_BlitScaled. The code in thst function had not changed since 5 years ago, so I assume whatever problems with it is not fixed.

However, I didn't observe anything off with the clipping. I also don't know what to look for.